### ISSUE TITLE
`calcite-checkbox` does not listen to keyboard interaction when screen reader (NVDA) is turned on

### ISSUE THREAD

**Issue by minhiu** [Esri/calcite-design-system#11996]
### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Summary

When using a screen reader (NVDA specifically), some calcite components does not listen to keyboard interaction.

This issue happens to Action, Button, and Checkbox

### Actual Behavior

When NVDA is on, the checkbox toggle state does not change by hitting "Enter" or "Space" key.

### Expected Behavior

When NVDA is on, the checkbox should be toggled by hitting "Enter" or "Space" key.

### Reproduction Sample

https://developers.arcgis.com/calcite-design-system/components/checkbox/

### Reproduction Steps

1. Turn on NVDA
2. Navigate to a checkbox
3. Hit "Enter" or "Space" key to toggle the checkbox
4. Notice the checkbox does not react to keyboard interaction

### Reproduction Version

3.1.0

### Priority impact

impact - p1 - need for current milestone

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Business/Community Analyst

**Comment by geospatialem** [comment]
Updated the above sample to reflect the difference between the native checkbox and Calcite checkbox via https://codepen.io/geospatialem/pen/YPXLKWg.

@minhiu This doesn't appear to affect either `button` or `action`, so removing their scope from the issue above. For instance, here's `button`: https://codepen.io/geospatialem/pen/azOGoWj, where the behavior is as-expected with JAWS and NVDA.

**Comment by geospatialem** [comment]
@driskull I'm not sure if there is anything we can do code-wise to improve this behavior, it seems like the behavior with AT is due to the `InteractiveContainer`/shadowDOM:

https://github.com/Esri/calcite-design-system/blob/46bb953e0482dccf53e96a04cb2bc2e55976ea98/packages/calcite-components/src/components/checkbox/checkbox.tsx#L245

Would any markup changes to accommodate AT behavior - without requiring the `Shift` key with `Enter`/`Space` to select the component be feasible?
